tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/metrics/CircuitBreakerMetricTest.java

```diff
@@ -0,0 +1,104 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.metrics;
+
+import static org.eclipse.microprofile.fault.tolerance.tck.metrics.Exceptions.expectCbOpen;
+import static org.eclipse.microprofile.fault.tolerance.tck.metrics.Exceptions.expectTestException;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.fault.tolerance.tck.metrics.CircuitBreakerMetricBean.Result;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.testng.annotations.BeforeTest;
+
+public class CircuitBreakerMetricTest extends Arquillian {
+
+    public static WebArchive deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "ftMetricCircuitBreaker.war")
+                .addClasses(CircuitBreakerMetricBean.class, MetricGetter.class);
+        return war;
+    }
+    
+    @Inject private CircuitBreakerMetricBean cbBean;
+    
+    @BeforeTest
+    public void closeTheCircuit() throws Exception {
+        // Assume the circuit is open
+        // Wait for it to half close, then close it with passing work
+        Thread.sleep(1500);
+        
+        for (int i = 0; i < 4; i++) {
+            cbBean.doWork(Result.PASS);
+        }
+    }
+    
+    @Test
+    public void testCircuitBreakerMetric() throws Exception {
+        MetricGetter m = new MetricGetter(CircuitBreakerMetricBean.class, "doWork");
+        m.baselineCounters();
+        
+        // First failure, circuit remains closed
+        expectTestException(() -> cbBean.doWork(Result.FAIL));
+        
+        assertThat(m.getCircuitBreakerState(), is(0));
+        assertThat(m.getCircuitBreakerCallsSuccessDelta(), is(0L));
+        assertThat(m.getCircuitBreakerCallsFailureDelta(), is(2L));
+        assertThat(m.getCircuitBreakerCallsCircuitOpenDelta(), is(0L));
+        assertThat(m.getCircuitBreakerOpenedDelta(), is(1L));
+
+        // Second failure, causes circuit to open
+        expectTestException(() -> cbBean.doWork(Result.FAIL));
+        
+        assertThat(m.getCircuitBreakerState(), is(2));
+        assertThat(m.getCircuitBreakerCallsSuccessDelta(), is(0L));
+        assertThat(m.getCircuitBreakerCallsFailureDelta(), is(2L));
+        assertThat(m.getCircuitBreakerCallsCircuitOpenDelta(), is(0L));
+        assertThat(m.getCircuitBreakerOpenedDelta(), is(1L));
```

Comment:
I think the 3rd request will trigger the circuit to open. The circuit will only be operated on a request though.